File: frontend/quiz-plateform/src/app/services/option.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { Observable } from 'rxjs';
import { Option } from '../models/option';

@Injectable({
  providedIn: 'root'
})
export class OptionService {

  private optionsUrl = `${baseUrl}/options`;

  constructor(private http: HttpClient) { }

  getOptionsByQuestion(questionId: number): Observable<Option[]> {
    return this.http.get<Option[]>(`${this.optionsUrl}/question/${questionId}`);
  }

  getOptionById(id: number): Observable<Option> {
    return this.http.get<Option>(`${this.optionsUrl}/${id}`);
  }

  createOption(option: Option): Observable<Option> {
    return this.http.post<Option>(this.optionsUrl, option);
  }

  updateOption(id: number, option: Option): Observable<Option> {
    return this.http.put<Option>(`${this.optionsUrl}/${id}`, option);
  }

  deleteOption(id: number): Observable<void> {
    return this.http.delete<void>(`${this.optionsUrl}/${id}`);
  }
}

File: backend/quizhire/quizhire/src/main/java/com/quizhire/quizhire/services/OptionService.java
package com.quizhire.quizhire.services;

import java.util.List;

import com.quizhire.quizhire.models.test.Option;
import com.quizhire.quizhire.models.test.Question;

public interface OptionService {
    List<Option> getOptionsByQuestion(Question question);
    Option getOptionById(Integer id) throws Exception;
    Option addOption(Option option);
    Option updateOption(Integer id, Option option);
    void deleteOption(Integer id) throws Exception;
}
